pyext/src/macros.py

```diff
@@ -552,7 +573,18 @@ def execute_macro(self):
             if save_frame:
                 print("--- frame %s score %s " % (str(i), str(score)))
 
-                if not self.test_mode:
+                if self.nest and not math.isnan(score):
+                    likelihood_for_sample = 1
+                    for rstrnt in self.nestor_restraints:
+                        likelihood_for_sample = (
+                            likelihood_for_sample * rstrnt.get_likelihood()
+                        )
```

Comment:
I think this would be more cleanly written as

    for rstrnt in self.nestor_restraints:
        likelihood_for_sample *= rstrnt.get_likelihood()

Reply:
Done.